docs/aws/audit/snsmonitoring/rules/is_topic_exposed.mdx

```diff
@@ -23,6 +23,229 @@ HITRUST, AWSWAF, SOC2, NISTCSF, PCIDSS
+
+2. Once AWS CLI is installed and configured, you can list all the SNS topics in your account by running the following command:
+
+   ```
+   aws sns list-topics
+   ```
+   This command will return a list of all SNS topics in your account.
+
+3. For each SNS topic, you can get the topic attributes which include the policy by running the following command:
+
+   ```
+   aws sns get-topic-attributes --topic-arn <TopicARN>
+   ```
+   Replace `<TopicARN>` with the ARN of the SNS topic. This command will return the attributes of the specified SNS topic.
+
+4. Check the policy of each SNS topic. If the policy allows the `sns:Publish` action for `Principal: "*"`, then the SNS topic is exposed. You can do this manually or write a script to automate the process. Here is a simple Python script that uses the `boto3` library to check for exposed SNS topics:
+
+   ```python
+   import boto3
+
+   sns = boto3.client('sns')
+
+   response = sns.list_topics()
+
+   for topic in response['Topics']:
+       attributes = sns.get_topic_attributes(TopicArn=topic['TopicArn'])
+       policy = attributes['Attributes']['Policy']
+       if 'sns:Publish' in policy and '"Principal":"*"' in policy:
+           print(f"Exposed SNS Topic: {topic['TopicArn']}")
+   ```
+   This script lists all SNS topics, gets the policy for each topic, and checks if the policy allows the `sns:Publish` action for `Principal: "*"`. If it does, it prints the ARN of the exposed SNS topic.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3): Boto3 makes it easy to integrate your Python application, library, or script with AWS services including AWS S3, AWS EC2, and more. You can install it using pip:
+
+```python
+pip install boto3
+```
+After installation, you need to configure it with your AWS credentials. You can do this in several ways, but the simplest is to use the AWS CLI:
+
+```bash
+aws configure
+```
+Enter your AWS Access Key ID, AWS Secret Access Key, Default region name, and Default output format when prompted.
+
+2. Import the necessary libraries and create a session:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    aws_session_token='SESSION_TOKEN',
+)
+```
+3. Use the SNS client to retrieve all the SNS topics:
+
+```python
+sns = session.client('sns')
+response = sns.list_topics()
+```
+4. Check the policy of each SNS topic to see if it's exposed:
+
+```python
+for topic in response['Topics']:
+    attributes = sns.get_topic_attributes(TopicArn=topic['TopicArn'])
+    policy = json.loads(attributes['Attributes']['Policy'])
+    for statement in policy['Statement']:
+        if statement['Effect'] == 'Allow' and statement['Principal'] == '*':
+            print(f"Exposed SNS Topic: {topic['TopicArn']}")
+```
+This script will print the ARN of all SNS topics that are exposed to the public.
+</Accordion>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>